server/server.js

```diff
@@ -1,15 +1,141 @@
+  "title": "The Coding Train",
+  "url": "https://www.youtube.com/watch?v=HerCR8bw_GE",
+  "rating": 230,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 82653,
+  "title": "Mac & Cheese | Basics with Babish",
+  "url": "https://www.youtube.com/watch?v=FUeyrEN14Rk",
+  "rating": 2111,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 858566,
+  "title": "Videos for Cats to Watch - 8 Hour Bird Bonanza",
+  "url": "https://www.youtube.com/watch?v=xbs7FT7dXYc",
+  "rating": 11,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 283634,
+  "title": "Learn Unity - Beginner's Game Development Course",
+  "url": "https://www.youtube.com/watch?v=gB1F9G0JXOo",
+  "rating": 211,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 562824,
+  "title": "Cracking Enigma in 2021 - Computerphile",
+  "url": "https://www.youtube.com/watch?v=RzWB5jL5RX0",
+  "rating": 111,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 442452,
+  "title": "Coding Adventure: Chess AI",
+  "url": "https://www.youtube.com/watch?v=U4ogK0MIzqk",
+  "rating": 671,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 536363,
+  "title": "Coding Adventure: Ant and Slime Simulations",
+  "url": "https://www.youtube.com/watch?v=X-iSQQgOd1A",
+  "rating": 76,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 323445,
+  "title": "Why the Tour de France is so brutal",
+  "url": "https://www.youtube.com/watch?v=ZacOS8NBK6U",
+  "rating": 73,
+  "time": "May 28th 2023, 8:47:44 pm"
+}];
+
 
 // GET "/"
 app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+  res.status(200).json({ data: videos });
 });
+
+
+// // POST "/add"
+app.post("/add", (req, res) => {
+  const { title, url } = req.body;
+  const regex1 = /(^https:\/\/www\.youtube\.com\/watch\?v=)([0-9a-zA-Z=\-_+.,:;'?!@#|$%^&*<>()]){11}$/;
+  const regex2 = /(^https:\/\/www.youtube.com\/shorts\/)([0-9a-zA-Z=\-_+.,:;'?!@#|$%^*<>()]){11}$/;
+  const result1 = regex1.test(url);
+  const result2 = regex2.test(url);
+  if (!result1 && !result2) {
+    res.status(400).json({
+      "result": "failure",
+      "message": "URL is not valid!"
+    });
+  }
+  else {
+    const id = Math.floor(Math.random() * 1000000);
+    const newVideo = { id, title, url, rating: 0, time: moment().format('MMMM Do YYYY, h:mm:ss a') }
+    videos.push(newVideo);
+    res.status(200).json({
+      "result": "success",
+      data: videos
+    });
+  }
+});
+
+
+// GET ONE "/search"
+app.get("/search/", (req, res) => {
+  const title = req.query.title;
+  const result = videos.filter(video => video.title.toLowerCase().includes(title.toLowerCase()));
+  if (!result)
+    res.status(400).json({
+      "result": "failure",
+      "message": "Video could not be found!"
+    })
+  else
+    res.status(200).json({ data: result })
+})
+
+
+// DELETE "/"
+app.delete("/:id", (req, res) => {
+  const id = parseInt(req.params.id);
```

Comment:
This can throw an error if id is not parseable as a number (e.g. `/not-a-number` would make `id = NaN` which will cause issues further down). It's good to add a `if` statement checking that id is not `NaN` (note, there's a function to do this, don't do `id != NaN` as JavaScript doesn't like it for historic reasons.